main.rb

```diff
@@ -0,0 +1,134 @@
+# rubocop:disable Metrics/ModuleLength,Metrics/CyclomaticComplexity,Style/CaseEquality,Metrics/PerceivedComplexity,Lint/RedundantCopDisableDirective
+
+# This module is a creation of the original Enumerable
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    a = self
+    a = *self if is_a? Range
+    a.length.times { |i| yield(a[i]) }
+    a
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    a = self
+    a = *self if is_a? Range
+    a.length.times { |i| yield(a[i], i) }
+    a
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    result = []
+    my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?(param = nil)
+    test_all = true
+
+    if !block_given? && param.nil?
+      my_each { |i| test_all = false if [false, nil].include? i }
+    elsif block_given? && param.nil?
+      my_each { |i| test_all = false unless yield(i) }
+    elsif param.is_a? Regexp
+      my_each { |i| test_all = false unless i =~ param }
+    elsif param.nil?
+      my_each do |i|
+        test_all = false unless i == true
+      end
+    elsif param.is_a? Class
+      my_each do |i|
+        test_all = false unless i.is_a?(param)
+      end
+      test_all
+    end
+    test_all
+  end
+
+  def my_any?(param = nil)
+    result = false
+    if block_given? && param.nil?
+      my_each { |i| return true if yield(i) }
+    elsif param.is_a? Regexp
+      my_each { |num| result = true if num =~ param }
+    elsif param.nil?
+      my_each do |i|
+        return true if i == true
+      end
+    elsif !param.nil?
+      my_each { |i| return true if i == param }
+    elsif param.is_a? Class
+      my_each do |i|
+        return true if i.is_a?(param)
+      end
+    end
+    result
```

Comment:
* [ ] `my_any?` is identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_any?(Integer) == array.any?(Integer) # false
  ```